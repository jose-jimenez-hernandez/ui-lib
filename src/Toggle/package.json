{
  "name": "@settle-ui/toggle",
  "version": "0.0.0",
  "description": "toggle description",
  "main": "../lib/toggle.es.js",
  "module": "../lib/toggle.es.js",
  "scripts": {
    "build": "vite build",
    "build:watch": "vite build --watch",
    "lint": "eslint ./src",
    "tsc": "tsc"
  },
  "types": "./src/toggle.d.ts",
  "dependencies": {
    "@settle-ui/spinner": "^0.0.0"
  }
}
