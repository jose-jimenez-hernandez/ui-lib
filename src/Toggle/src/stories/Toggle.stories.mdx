import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';

| Category   | Type    | Title  |
| ---------- | ------- | ------ |
| Components | General | Toggle |

Switching Selector.

## When To Use
- If you need to represent the switching between two states or on-off state.
- The difference between `Switch` and `Checkbox` is that `Switch` will trigger a state change directly when you toggle it, 
while `Checkbox` is generally used for state marking, which should work in conjunction with submit operation.


## API

| Property | Description                                           | Type                                   | Default    | 
| -------- | ----------------------------------------------------- | -------------------------------------- | ---------- | 
| checked | Determine whether the Switch is checked | boolean | false |
| className | The additional class to Switch | string | - |
| disabled | Disable switch | boolean | false |
| Icon | Show Icon | boolean | false  |
| loading | Loading state of switch | boolean | false |
| size | The size of the Switch, options: `default` `small` | string | `default` |

It accepts all props which native support.
