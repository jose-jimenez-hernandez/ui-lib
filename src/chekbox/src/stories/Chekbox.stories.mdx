import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';

| Category   | Type    | Title  |
| ---------- | ------- | ------ |
| Components | General | Chekbox |

To trigger an operation.

## When To Use



## API

| Property | Description                                           | Type                                   | Default    | 
| -------- | ----------------------------------------------------- | -------------------------------------- | ---------- | 

It accepts all props which native support.
