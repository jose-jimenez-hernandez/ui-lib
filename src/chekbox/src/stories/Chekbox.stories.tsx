import React from 'react';
import { Meta } from '@storybook/react/types-6-0';
import { Story } from '@storybook/react';
import Chekbox from '../Chekbox';
import ChekboxDoc from './Chekbox.stories.mdx';

export default {
  title: 'Components/Chekbox',
  component: Chekbox,
  parameters: {
    docs: {
      page: (ChekboxDoc as any).parameters?.docs.page,
    },
  },
};

const ChekboxTemplate: Story = (args: any) => ( <Chekbox  {...args}/>
);

export const DefaultChekbox = ChekboxTemplate.bind({});
DefaultChekbox.args = {  };
