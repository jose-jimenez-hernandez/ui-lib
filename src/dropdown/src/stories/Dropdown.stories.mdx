import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';

| Category   | Type    | Title  |
| ---------- | ------- | ------ |
| Components | General | Dropdown |

To trigger an operation.

## When To Use



## API

| Property | Description                                           | Type                                   | Default    | 
| -------- | ----------------------------------------------------- | -------------------------------------- | ---------- | 

It accepts all props which native support.
