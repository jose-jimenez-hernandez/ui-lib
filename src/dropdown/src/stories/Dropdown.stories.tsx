import React from 'react';
import { Meta } from '@storybook/react/types-6-0';
import { Story } from '@storybook/react';
import Dropdown from '../Dropdown';
import DropdownDoc from './Dropdown.stories.mdx';
import DropdownItem from '../DropdownItem';

const items = [
  { id: '1', name: 'Wade Cooper' },
  { id: '2', name: 'Arlene Mccoy' },
  { id: '3', name: 'Devon Webb' },
  { id: '4', name: 'Tom Cook' },
  { id: '5', name: 'Tanya Fox' },
  { id: '6', name: 'Hellen Schmidt' },
  { id: '7', name: 'Caroline Schultz' },
  { id: '8', name: 'Mason Heaney' },
  { id: '9', name: 'Claudie Smitham' },
  { id: '10', name: 'Emil Schaefer' },
];

export default {
  title: 'Components/Dropdown',
  component: Dropdown,
  parameters: {
    docs: {
      page: (DropdownDoc as any).parameters?.docs.page,
    },
  },
};

const DropdownTemplate: Story = (args: any) => (
  <div className="flex justify-end">
    <Dropdown {...args}>
      {items &&
        items.map((item) => (
          <DropdownItem key={item.id} label={item.name} separator={true} value={item}></DropdownItem>
        ))}
    </Dropdown>
  </div>
);

export const DefaultDropdown = DropdownTemplate.bind({});
DefaultDropdown.args = {};
