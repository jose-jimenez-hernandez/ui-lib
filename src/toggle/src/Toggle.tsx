import { useEffect, useState } from 'react';
import { Switch } from '@headlessui/react';
import { ToggleProps } from './ToggleType';
import useStyles from './styles';

const Toggle = ({ className, disabled, label, variant, size, checked, icon, onChange }: Partial<ToggleProps>) => {
  const [enabled, setEnabled] = useState(checked);
  const styles = useStyles({
    disabled,
    label,
    variant,
    className,
    size,
  });

  useEffect(() => {
    setEnabled(Boolean(enabled));
  }, [checked]);

  const onChangeHandle = (checked: boolean) => {
    onChange?.(!checked);
    setEnabled(Boolean(!enabled));
  };

  return (
    <Switch.Group as="div" className={styles.container}>
      <Switch
        checked={Boolean(enabled)}
        disabled={disabled}
        onChange={onChangeHandle}
        className={styles.switch(Boolean(enabled))}
      >
        <span className="sr-only">Use setting</span>
        <span aria-hidden="true" className={styles.switchAnimation} />
        <span aria-hidden="true" className={styles.tracker(enabled)} />
        <span aria-hidden="true" className={styles.dragger(enabled)}>
          {icon && (
            <>
              <span className={styles.iconOff(enabled)} aria-hidden="true">
                <svg className="h-3 w-3 text-gray-400" fill="none" viewBox="0 0 12 12">
                  <path
                    d="M4 8l2-2m0 0l2-2M6 6L4 4m2 2l2 2"
                    stroke="currentColor"
                    strokeWidth={2}
                    strokeLinecap="round"
                    strokeLinejoin="round"
                  />
                </svg>
              </span>
              <span className={styles.iconOn(enabled)} aria-hidden="true">
                <svg className="h-3 w-3 text-indigo-600" fill="currentColor" viewBox="0 0 12 12">
                  <path d="M3.707 5.293a1 1 0 00-1.414 1.414l1.414-1.414zM5 8l-.707.707a1 1 0 001.414 0L5 8zm4.707-3.293a1 1 0 00-1.414-1.414l1.414 1.414zm-7.414 2l2 2 1.414-1.414-2-2-1.414 1.414zm3.414 2l4-4-1.414-1.414-4 4 1.414 1.414z" />
                </svg>
              </span>
            </>
          )}
        </span>
      </Switch>
      <Switch.Label as="span" className={styles.switchLabel}>
        {label}
      </Switch.Label>
    </Switch.Group>
  );
};

export default Toggle;
