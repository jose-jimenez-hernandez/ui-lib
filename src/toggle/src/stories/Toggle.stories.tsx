import { Story } from '@storybook/react';
import Toggle from '../Toggle';
import ToggleDoc from './Toggle.stories.mdx';
import { SIZE } from '../@settle-ui/utils';

export default {
  title: 'Components/Toggle',
  component: Toggle,
  parameters: {
    docs: {
      page: (ToggleDoc as any).parameters?.docs.page,
    },
  },
};

const hideControls = {
  controls: { exclude: /.*/, hideNoControlsWarning: true },
};

const ToggleTemplate: Story = (args: any) => <Toggle {...args} />;

export const $PlayWithMe = ToggleTemplate.bind({});

export const StateOn = ToggleTemplate.bind({});
StateOn.args = { value: true };
StateOn.parameters = hideControls;

export const StateDisabled = ToggleTemplate.bind({});
StateDisabled.args = { disabled: true };
StateDisabled.parameters = hideControls;

export const Label = ToggleTemplate.bind({});
Label.args = { label: 'Toggle label' };
Label.parameters = hideControls;

const ToggleSizeTemplate: Story = () => (
  <div className="flex gap-4">
    <Toggle label="small" size={SIZE.SM} />
    <Toggle label="default" size={SIZE.BASE} />
  </div>
);

export const Size = ToggleSizeTemplate.bind({});
Size.parameters = hideControls;

const ToggleIconTemplate: Story = () => (
  <div className="flex gap-4">
    <Toggle label="switch on" icon={true} checked={true} />
    <Toggle label="switch off" icon={true} />
  </div>
);

export const Icons = ToggleIconTemplate.bind({});
Icons.parameters = hideControls;
