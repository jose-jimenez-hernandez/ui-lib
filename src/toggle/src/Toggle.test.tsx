import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { composeStories } from '@storybook/testing-react';
import * as stories from './stories/Toggle.stories';

const { $PlayWithMe: DefaultToggle } = composeStories(stories);

it('renders correctly', () => {
  const state = render(<DefaultToggle />).baseElement;

  expect(state).toMatchSnapshot();
});
