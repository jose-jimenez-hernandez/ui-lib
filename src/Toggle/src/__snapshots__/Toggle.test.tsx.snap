// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<body>
  <div>
    <div
      class="flex items-center"
    >
      <button
        aria-checked="false"
        aria-labelledby="headlessui-label-2"
        class="bg-gray-200 flex-shrink-0 group relative rounded-full inline-flex items-center justify-center cursor-pointer  focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500 h-6 w-11"
        id="headlessui-switch-1"
        role="switch"
        tabindex="0"
        type="button"
      >
        <span
          class="sr-only"
        >
          Use setting
        </span>
        <span
          aria-hidden="true"
          class="pointer-events-none absolute bg-white w-full h-full rounded-md"
        />
        <span
          aria-hidden="true"
          class="bg-gray-200 pointer-events-none absolute  mx-auto rounded-full transition-colors ease-in-out duration-200 h-6 w-11"
        />
        <span
          aria-hidden="true"
          class="translate-x-0 pointer-events-none absolute left-0 inline-block h-5 w-5 border border-gray-200 rounded-full bg-white shadow transform ring-0 transition-transform ease-in-out duration-200"
        />
      </button>
      <span
        class="text-sm font-medium text-gray-900 ml-3"
        id="headlessui-label-2"
      />
    </div>
  </div>
</body>
`;
